### PR TITLE
Add instructions for footer donations

### PR DESCRIPTION
**Pull request by alkis**
Add instructions for footer donations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,28 @@
...
+inspection before submission.
+
+When you are ready to donate a footer please open a PR against this repository
+and add your footer under `footer/<name>.footer`.
+
+Use `parquet-dump-footer --help` for explantion of all the options.
+
+## alternate parquet-dump-footer binary
+
+You can binaries in this repo for different architectures. The binaries are built
```
Comment (rok) [inline README.md]:
```suggestion
You can use binaries in this repo for different architectures. The binaries are built
```

File: README.md [hunk 2]
```diff
@@ -1 +1,28 @@
...
+using the following cmake configuration.
```
Comment (rok) [inline README.md]:
```suggestion
from parquet-cpp using the following cmake configuration.
```

File: README.md [hunk 3]
```diff
@@ -1 +1,28 @@
...
+
+```sh
```
Comment (rok) [inline README.md:27]:
```suggestion
```sh
cd arrow/cpp/build
```

**Review by rok — COMMENTED** [review]
This looks great. Suggesting some changes to make it clearer that `parquet-dump-footer` source is within `parquet-cpp` repo.

File: README.md [hunk 4]
```diff
@@ -1 +1,29 @@
...
+
+When you are ready to donate a footer please open a PR against this repository
+and add your footer under `footer/<name>.footer`.
+
+Use `parquet-dump-footer --help` for explantion of all the options.
+
+## alternate parquet-dump-footer binary
+
+You can use binaries in this repo for different architectures. The binaries are built
+from parquet-cpp using the following cmake configuration.
```
Comment (rok) [inline README.md]: Perhaps a link to the source?
```suggestion
from [parquet-cpp](https://github.com/apache/arrow/blob/main/cpp/tools/parquet/parquet_dump_footer.cc) using the following cmake configuration.
```

**Review by rok — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1 +1,29 @@
-# Apache Parquet Benchmarking
+# Parquet benchmark data
```
Comment (vinooganesh) [inline README.md:1]:
```suggestion
# Apache Parquet Benchmark Data
```

File: README.md [hunk 6]
```diff
@@ -1 +1,29 @@
...
+of parquet tools. This utility extracts footers from parquet, scrubs binary data
+for privacy reasons and allows to pretty print (`--debug`) the result for
+inspection before submission.
+
+When you are ready to donate a footer please open a PR against this repository
+and add your footer under `footer/<name>.footer`.
+
+Use `parquet-dump-footer --help` for explantion of all the options.
+
+## alternate parquet-dump-footer binary
```
Comment (vinooganesh) [inline README.md:21]:
```suggestion
## Alternate parquet-dump-footer binary
```

**Review by vinooganesh — APPROVED** [review]

**Review by steveloughran — COMMENTED** [review]
I'd prefer the binaries a release artifact rather than something in the source tree.

It'd be good to show example output to reassure people about what is extracted; maybe make reviewing the contents of a footer part of the PR process

on the topic of process, is there any way to set up a CI validation that a footer is well formed?

**Comment by alkis** [comment]
> I'd prefer the binaries a release artifact rather than something in the source tree.

This is a stop gap solution until we get the next release of parquet tools released.

**Comment by alkis** [comment]
Friendly ping, can we merge this?

**Comment by pitrou** [comment]
Hmm, perhaps we should put all of this in a `footers` subdirectory? This repo may be useful for other kinds of data at some point.

**Comment by pitrou** [comment]
Also, can the binaries be compressed somehow?

**Comment by alkis** [comment]
> Also, can the binaries be compressed somehow?

Are you referring to the executables? I can compress them but then they become less convenient to run. I would rather keep the convenience.

**Comment by pitrou** [comment]
Yes, I'm referring to the executables. It's better to avoid large files in git repos IMHO.

**Comment by alkis** [comment]
I tried they compressed to 22mb (if I add both in a zip file).

**Comment by alkis** [comment]
Ok I put them in a zip.

File: README.md [hunk 7]
```diff
@@ -1 +1,28 @@
...
+
+This repository contains Parquet benchmark data. Such data is useful to help
+optimize Parquet implementations but also advance the Parquet format itself.
+
+At this point the community requests donation of Parquet footers and especially
+footers that are large and slow to parse/process. Typically these are footers of
+wide schemata: either coming from lots of individual columns and/or deeply nested
+structs.
+
+To donate Parquet footers we have built a binary `parquet-dump-footer` as part
```
Comment (emkornfield) [inline README.md:11]: could you add a note that this is now compressed?

**Review by emkornfield — APPROVED** [review]
One small comment on making it clear that the footer is now binary.

**Comment by emkornfield** [comment]
@julienledem let me know if you still want to review, otherwise I'll merge tonight or tomorrow.

**Review by julienledem — APPROVED** [review]
Thank you for addressing al the comments. This looks good to me.

**Comment by julienledem** [comment]
> @julienledem let me know if you still want to review, otherwise I'll merge tonight or tomorrow.

I just reviewed (but don't need/want to be a bottleneck), please merge, thank you!

**Pull request merged**